azure.go

```diff
@@ -279,32 +332,29 @@ func (ac *AzureClient) listFromResourceGroup(resourceGroup string, resourceTypes
 		return nil, fmt.Errorf("Error: %v", err)
 	}
 	defer resp.Body.Close()
+	body, err := ioutil.ReadAll(resp.Body)
+
 	if resp.StatusCode != 200 {
-		return nil, fmt.Errorf("Unable to query resource group API with status code: %d", resp.StatusCode)
+		return nil, fmt.Errorf("Unable to query resource API with status code: %d with body: %s", resp.StatusCode, body)
 	}
 
-	body, err := ioutil.ReadAll(resp.Body)
 	if err != nil {
 		return nil, fmt.Errorf("Error reading body of response: %v", err)
 	}
+	return body, err
 
-	var data AzureResourceListResponse
-	err = json.Unmarshal(body, &data)
-	if err != nil {
-		return nil, fmt.Errorf("Error unmarshalling response body: %v", err)
-	}
+}
 
+func extractMetricNames(data AzureResourceListResponse, subscription string) []string {
```

Comment:
Are you sure you didn't mean:
```suggestion
// Extract resource names from the AzureResourceListResponse
func extractResourceNames(data AzureResourceListResponse, subscription string) []string {
```